framework/utils/CPropertyValue.php

```diff
@@ -53,7 +53,7 @@ class CPropertyValue
 	public static function ensureBoolean($value)
 	{
 		if (is_string($value))
-			return !strcasecmp($value,'true') || $value!=0;
+			return !strcasecmp($value,'true') || ($value!=0 && $value!=='' && is_numeric($value));
```

Comment:
@samdark Do you have examples of `$value` that are fixed with this change? Just want to make sure this change is correct.

Reply:
@marcovtwout revert this like in master and run tests on PHP 8. That's covered with tests. It is about https://wiki.php.net/rfc/string_to_number_comparison